satpy/readers/iasi_l2.py

```diff
@@ -105,62 +113,69 @@ def get_dataset(self, key, info):
         with h5py.File(self.filename, 'r') as fid:
             LOGGER.debug('Reading %s.', key.name)
             if key.name in DSET_NAMES:
-                m_data = read_dataset(fid, key, info)
+                m_data = read_dataset(fid, key)
             else:
-                m_data = read_geo(fid, key, info)
-        m_data.info.update(info)
+                m_data = read_geo(fid, key)
+        m_data.attrs.update(info)
 
         return m_data
 
 
-def read_dataset(fid, key, info):
+def read_dataset(fid, key):
     """Read dataset"""
     dsid = DSET_NAMES[key.name]
-    data = fid["/PWLR/" + dsid].value
+    dset = fid["/PWLR/" + dsid]
+    if dset.ndim == 3:
+        dims = ['y', 'x', 'level']
+    else:
+        dims = ['y', 'x']
+    data = xr.DataArray(da.from_array(dset.value, chunks=CHUNK_SIZE),
+                        name=key.name, dims=dims).astype(np.float32)
+    data = xr.where(data > 1e30, np.nan, data)
```

Comment:
I don't know if there is a performance difference, but FYI there is also a `data.where(data <= 1e30)` which will fill anything with NaN where the condition is False. But what you have is probably equivalent under the hood.

Reply:
Good to know, although I prefer to have the chosen data shown explicitly. At least for this is more understandable.